dut/drivers.wake

```diff
@@ -0,0 +1,107 @@
+# LocalMetalDirectory => the metal/drivers directory in the device's repository
+global def makeBasicDriverPlan duhFile omFile vendorName deviceName localMetalDirectory dutName =
+  BasicDriverPlan duhFile omFile vendorName deviceName localMetalDirectory dutName True False
+
+# This function provides an external interface through the publish/subscribe
+# variable 'basicDriverPlans' If you want automatic basic driver creation,
+# create a BasicDriverPlan using makeBasicDriverPlan, Then build the drivers
+# by calling createBasicDriver
+def getDevicePlan dutName =
+  def plans = subscribe basicDriverPlans
+  def samePlan aPlan =
+    aPlan.getBasicDriverPlanDUTName ==~ dutName
+  find samePlan plans
+  | omap getPairFirst
+
+def basicPlanToPlan basicDriverPlan =
+  def omfile = basicDriverPlan.getBasicDriverPlanOMFile Unit
+  def extraFlags =
+    def overwrite =
+      if basicDriverPlan.getBasicDriverPlanOverwriteExisting
+      then "--overwrite-existing", Nil
+      else Nil
+    def onlyBaseHeader =
+      if basicDriverPlan.getBasicDriverPlanOnlyBaseHeader
+      then "--base-header-only", Nil
+      else Nil
+    overwrite ++ onlyBaseHeader
+  def cmd =
+    "{here}/../scripts/generate_drivers.py".simplify,
+    "--object-model",
+    omfile.getPathName,
+    "--duh-document",
+    basicDriverPlan.getBasicDriverPlanDUHDocument.getPathName,
+    "--vendor",
+    basicDriverPlan.getBasicDriverPlanVendorName,
+    "--device",
+    basicDriverPlan.getBasicDriverPlanDeviceName,
+    "--metal-dir",
+    basicDriverPlan.getBasicDriverPlanLocalMetalDirectory,
+    extraFlags
+  def visibleFiles =
+    "{here}/../scripts/generate_drivers.py".simplify.source,
+    basicDriverPlan.getBasicDriverPlanDUHDocument,
+    omfile,
+    basicDriverPlan.getBasicDriverPlanLocalMetalDirectory.source,
+    Nil
+  makePlan cmd visibleFiles
+  | addPythonEnv ("{here}/../scripts/generate_drivers_env".simplify)
+
+tuple DriverOutput =
+  global CFiles:      List Path
+  global IncludeDirs: List String
+
+def foo x =
+  def bar = format x | println
+  x
+
+def makeDriverOutput basicDriverPlan jobOutputs =
+  def ifPathCFile path =
+    path.getPathName
+    | matches `.*\.[c,h]`
+  def cFiles = filter ifPathCFile jobOutputs
+  def metalDir = basicDriverPlan.getBasicDriverPlanLocalMetalDirectory
+  def deviceName = basicDriverPlan.getBasicDriverPlanDeviceName
+  def includeDirs =
+    metalDir,
+    Nil
+  DriverOutput cFiles includeDirs
+
+
+def makeBasicDrivers basicDriverPlan =
+  basicPlanToPlan basicDriverPlan
+  | runJob
+  | getJobOutputs
+  | makeDriverOutput basicDriverPlan
+
+# External wake target to create basic drivers
+# deviceName => the name of the DUT plan
+global def createBasicDriver dutPlanName =
+  getDevicePlan dutPlanName
+  | omap makeBasicDrivers
+
+# wake target to create basic drivers from internal code where
+# omfile is generated
+# deviceName => the name of the DUT Plan
+global def createBasicDriverWithOMFile dutPlanName omfile =
```

Comment:
It's not clear to me when you would use this function vs the `createBasicDriver` function. What is the difference in terms of the output between the two?

Reply:
They are basically the same. CreateBasicDriverWithOMFile is used when you actually have an omfile. But, it's just a convenience function really to deal with the omfile being delayed.